libs/components/config/src/lib/params.ts

```diff
@@ -190,7 +190,9 @@ export class SkyAppRuntimeConfigParams {
     // Combine all parameters and their values, e.g. 'a=b'.
     const joinedParams = httpParams
       .keys()
-      .map((key) => `${key}=${httpParams.get(key)}`);
+      // Account for parameters with multiple values.
+      .map((key) => httpParams.getAll(key)?.map((value) => `${key}=${value}`))
```

Comment:
You're not encoding the key or value here. Calling `getUrlSearchParams()` which creates a new instance of `HttpParams` is likely going to decode the keys and values, so they'd need to be re-encoded here. This appears to also be an issue with SPA parameter names. You should be able to test it like this:

```
  it('should preserve query string component encoding', () => {
    const params = new SkyAppRuntimeConfigParams('?%3F%26%26=%26%26%3F', allowed);

    expect(params.getUrl('https://mysite.com?%3F%26=%26%3F')).toEqual(
      'https://mysite.com?%3F%26=%26%3F&%3F%26%26=%26%26%3F'
    );
  });
```

Fixing this may be as simple as updating line 185 to not encode the value, then calling `httpParams.toString()` which will encode the keys and values for you. https://angular.io/api/common/http/HttpParams